lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -145,33 +145,35 @@ final class DocParser
      */
     private static $annotationMetadata = [
         Annotation\Target::class => [
-            'is_annotation'    => true,
-            'has_constructor'  => true,
-            'properties'       => [],
-            'targets_literal'  => 'ANNOTATION_CLASS',
-            'targets'          => Target::TARGET_CLASS,
-            'default_property' => 'value',
-            'attribute_types'  => [
+            'is_annotation'         => true,
+            'has_constructor'       => true,
+            'has_named_constructor' => false,
```

Comment:
This might be confusing: to me, a named constructor is a static method that returns `self`

Reply:
You are right. I wanted to keep it short since it's only an internal representation, but I understand that this name causes confusion. I'll change it.

Reply:
@derrabus ok i think we will take this approach here, so that leaves renaming this option as the last todo if @greg0ire agrees. I would merge afterwards.

Reply:
Done.